### PR TITLE
add : home page with notes functionality

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I've added Home page with basic functionalities like Create, Edit and Delete note.

### Files to check:
- Home.jsx
- NoteCard.jsx
- EditModal.jsx
- Sidebar.jsx
- files from routes folder
- CSS files

### Please review this PR and give your feedback

### CODE REVIEW

File: src/components/sidebar/sidebar.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#header .pro-sidebar-inner .pro-sidebar-layout ul {
+    padding: 0 5px;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout ul .pro-inner-item {
+    color: #000;
+    margin: 10px 0px;
+    font-weight: bold;
+}
+#header .pro-sidebar-inner .pro-sidebar-layout ul .pro-inner-item .pro-icon-wrapper {
+    background-color: transparent;
```
Comment (Maianki) [inline src/components/sidebar/sidebar.css:34]: Suggestion: Avoid using so many nesting of ID and classes. It can lead to specificity war.

**Review by Maianki — APPROVED** [review]
Great work Apurva. Your code looks good and follows good naming conventions. You are using BEM so great work there too. You can go ahead with the merge.

File: src/components/sidebar/sidebar.css (hunk 1 above)
Comment (ApurvaSawant11) [reply src/components/sidebar/sidebar.css:34]: I am using `react-pro-sidebar`. To change their default CSS this nesting is necessary.

File: src/context/note-context.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  const value = {
+    getNotes,
+    postNotes,
+    editNote,
+    deleteNote,
+    noteList: state.noteList,
+    state,
+    dispatch,
+  };
+  return <NoteContext.Provider value={value}>{children}</NoteContext.Provider>;
```
Comment (Developers-codz) [inline src/context/note-context.jsx:104]: Appreciations - Liked the way u made a variable for list of values

File: src/pages/home/home.css [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+/* || Overriding react-quill properties */
+
+.quill {
+    position: relative;
+    height: max-content !important;
+}
+
+.ql-editor {
+    padding-top:2px !important ;
```
Comment (Developers-codz) [inline src/pages/home/home.css:58]: using important is not suggestible

**Review by Developers-codz — APPROVED** [review]
UI is amazing, codes  are also well structured and readable, good work. Just few observations

File: src/pages/home/home.css (hunk 3 above)
Comment (ApurvaSawant11) [reply src/pages/home/home.css:58]: This if for overriding `react-sidebar` styles. That's why I am using `!important`

**Pull request merged**